File: Chapter-2/hello.js
// node hello.js

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('What is your name? ', (name) => {
  console.log('Hello, %s, nice to meet you!', name)
  rl.close();
});

File: Chapter-2/hello.rs
use std::io::prelude::*;                                                           
use std::io;

fn main() {
  let mut buffer = String::new();

  print!("What is your name? ");
  io::stdout().flush().unwrap();

  io::stdin().read_line(&mut buffer).unwrap();
  let name = buffer.trim_right();

  println!("Hello, {}, nice to meet you!", name);
}
